src/main/java/newbank/server/Account.java

```diff
@@ -2,24 +2,80 @@
 
 import org.javamoney.moneta.Money;
 
+import newbank.server.exceptions.AccountBalanceInvalidException;
+import newbank.server.exceptions.AccountInvalidException;
 import newbank.server.exceptions.AccountNameInvalidException;
 
+/**
+ * Representation of a customer's account.
+ *
+ * <p>An account is identified by a name and has a balance. An account also has a flag indicating
+ * whether that account is the default current account for a customer. The default current account
+ * is the account used (by default) for all transactions (for instance when receiving money from
+ * another customer or when paying money to a customer). Note that a Savings account cannot be used
+ * as the default current account.
+ */
 public class Account {
   private String accountName;
   private Money balance;
+  private boolean isDefault = false;
 
   public static final int MIN_NAME_LENGTH = 4;
   public static final int MAX_NAME_LENGTH = 12;
 
-  public Account(String accountName, Money openingBalance) throws AccountNameInvalidException {
-    this.validate(accountName, openingBalance);
+  public static class Builder {
+    private String accountName;
+    private Money balance;
+    private boolean isDefault;
 
-    this.accountName = accountName;
-    this.balance = openingBalance;
+    public Builder(final String accountName) throws AccountNameInvalidException {
+      validateName(accountName);
+
+      this.accountName = accountName;
+    }
+
+    public Builder openingBalance(final Money openingBalance)
+        throws AccountBalanceInvalidException {
+      validateBalance(openingBalance);
+
+      this.balance = openingBalance;
+
+      return this;
+    }
+
+    public Builder asDefault() throws AccountInvalidException {
```

Comment:
Perhaps it should be just the `Customer` class that decides the default account? I don't think it's needed to store this information per each `Account`. The current design is very limiting in terms of time complexity if we ever were to support a big number of account for bigger customers.

Reply:
Good point. Can be done. I'll try to refactor.

Kindest Regards,
Jules